### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by caherrerapa** [hanhdt/AndroidTVChannels#1]
Hi
Here some comments regarding the code
1.  The api code is in the activity and this could be better and cleaner normally via a service. I haven't used volley much but it kinda feels like you should move to a service in the application class rather than coding a lot inside the fragments. 

please check this out: http://stackoverflow.com/a/18335433
1.   You also define Volley code in the Application class so it's quite hard to follow the structure. The code is functional but needs more thought in how to organize the different elements
2.  There should be a clean way to process the reponse obtained from the api different to get using keys, e.g. Gson and request interface?
3. The api key should go on the resource file and the api service or equivalent should be initialized there.
4. To add the api key i would be expecting to see an request interceptor for all api requests that adds the api_key to the headers or to the body

In general the code works but it's hard to follow as it has different pieces glued together in a not so organized way using highly tied components that are hard to change or to test in case we wanna do unit testing.

This seems like a good starting point to organize things using volley https://yakivmospan.wordpress.com/2014/03/04/volley-request-manager/

**Comment by hanhdt** [comment]
Hi, 

Thank you for your helpful comments. I will read all. 

Also, in my mind, I have planned to refactoring and improving organize things for Volley.

Regards.

**Comment by hanhdt** [comment]
Hi,

I organized code's structure follow to your suggestions. 

Ideally, I applied Design Patterns that are improve to flexible, maintainable,...

Thank you for your reviews.
